### DISCUSSION TITLE
Display Full App Title Information

### DISCUSSION THREAD

**Discussion by itaysharir** [FelixKratz/SketchyBar#245] [Q&A] [answered]
How can i display the app's title instead of just the app's name?

For example:
**vim ~/.config/sketchybar/sketchybarrc**
instead of:
**Alacritty**

Thanks in advance!

**Comment by FelixKratz** [comment, marked as answer]
First of all, ask as many questions as you like, because questions built up the Q&A section and make it easier for other people to find answers to their questions as well.

I have this solution lurking on my hard drive which uses yabai:

sketchybarrc:
```bash
sketchybar --add       event        window_focus                         \
           --add       event        title_change                         \
                                                                         \
           --add       item         window_title left                    \
           --set       window_title script="$PLUGIN_DIR/window_title.sh" \
                                    icon.drawing=off                     \
           --subscribe window_title window_focus title_change
```

$PLUGIN_DIR/window_title.sh:
```bash
#!/usr/bin/env sh

TITLE=$(yabai -m query --windows --window | jq -r '.title')
sketchybar --set $NAME label="$TITLE"
```

yabairc:
```bash
yabai -m signal --add event=window_focused action="sketchybar --trigger window_focus"
yabai -m signal --add event=window_title_changed action="sketchybar --trigger title_change"
```

You could decide to cut the window title if it surpasses a certain length, or add more window informations like e.g. shown here:
https://github.com/FelixKratz/SketchyBar/discussions/12#discussioncomment-1633997

**Comment by itaysharir** [comment]
Thanks!
